src/interfaces/token.ts

```diff
@@ -35,17 +35,25 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
    * @param overrides
    * @returns Usdc exchange rate (6 decimals)
    */
-  async priceUsdc<T extends Address>(token: T, overrides?: CallOverrides): Promise<Usdc>;
+  async priceUsdc<T extends Address>(token: T, overrides?: CallOverrides): Promise<Usdc | null>;
 
   /**
    * Get the suggested Usdc exchange rate for list of tokens.
    * @param tokens
    * @param overrides
    * @returns Usdc exchange rate map (6 decimals)
    */
-  async priceUsdc<T extends Address>(tokens: T[], overrides?: CallOverrides): Promise<TypedMap<T, Usdc>>;
+  async priceUsdc<T extends Address>(tokens: T[], overrides?: CallOverrides): Promise<TypedMap<T, Usdc> | null>;
+
+  async priceUsdc<T extends Address>(
+    tokens: T | T[],
+    overrides?: CallOverrides
+  ): Promise<TypedMap<T, Usdc> | Usdc | null> {
+    if (!Chains[this.chainId]) {
+      console.error(`the chain ${this.chainId} hasn't been implemented yet`);
+      return null;
```

Comment:
Lets throw an error here better since its probably better to let the user know, and also we maintain the same return interface.